### PR TITLE
SolidusAssetsProxy

### PR DESCRIPTION
**Pull request by joanniclaborde**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+var _ = require('underscore');
+
+var SolidusAssetsProxy = function(proxied_assets) {
+  if (!(this instanceof SolidusAssetsProxy)) return new SolidusAssetsProxy(proxied_assets);
+
+  this.proxied_assets = proxied_assets.map(function(assets) {
+    // The origin RegExp will try to match URLs like this:
+    // (staring delimiter)(origin + everything up to ending delimiter)(ending delimiter)
+    // For example: (")(http://some/url/.*?)(")
```
Comment (localjo) [inline index.js]: How about just saying "Match URL strings from protocol up to the query string"?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
+# SolidusAssetsProxy
+
+This [Solidus](https://github.com/solidusjs/solidus) library provides an easy solution for proxying a site's assets with a CDN or a an image processing service like [imgix](https://www.imgix.com/) and [ReSRC](https://www.resrc.it/).
```
Comment (pushred) [inline README.md]: ReSRC was actually acquired by Fastly and the service is gonna be merged into it somehow. Just imgix is fine as an example.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+```javascript
+var SolidusAssetsProxy = require('solidus-assets-proxy');
+var assets_proxy = new SolidusAssetsProxy([
+  {
+    origin: 'http://resource1.com/images',
+    proxy: 'http://mysite.proxy1.com'
+  },
+  {
+    origin: /http:\/\/resource(?:2|3)\.com\/pictures/, // Note the non-capturing group
+    proxy: 'http://proxy2.com/mysite'
```
Comment (pushred) [inline README.md]: This seems a bit at odds with the statement "A single redirects server can be used by multiple Solidus sites, and even when developing a new site" that comes right before. Why not:

``` js
new SolidusAssetsProxy({
  origins: ['http://resource1.com/images'],
  proxy: 'http://mysite.proxy1.com'
});
```

Deferring varying proxies until a valid use case for that materializes.

File: index.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+module.exports = SolidusAssetsProxy;
+
+// PRIVATE
+
+function escapeRegExp(string) {
+  return string.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
+};
+
+function addSlash(string) {
+  return string.slice(-1) == '/' ? string : string + '/';
```
Comment (pushred) [inline index.js]: Doesn't [`url.resolve`](https://nodejs.org/api/url.html#url_url_resolve_from_to) do this?

File: index.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+};
+
+function proxyObjectAssetsByPath(object, path) {
+  if (!object) return;
+  if (_.isArray(object)) {
```
Comment (pushred) [inline index.js:47]: Any reason you're not using `Array.isArray` beyond uniformity with the other checks?

File: README.md (hunk 3 above)
Comment (joanniclaborde) [inline README.md]: Good idea, and if we ever need multiple proxies, then multiple `SolidusAssetsProxy` instances can also be used.

File: index.js (hunk 4 above)
Comment (joanniclaborde) [inline index.js]: Yes!!

File: index.js (hunk 5 above)
Comment (joanniclaborde) [inline index.js:47]: Nope, uniformity.

File: index.js (hunk 4 above)
Comment (joanniclaborde) [inline index.js]: Err I mean No!! I would need `url.join` instead...

**Comment by joanniclaborde** [comment]
Alright all fixed!

**Comment by pushred** [comment]
:+1: over here

**Pull request merged**